### PR TITLE
JDK-8260335: [macos] Running app using relative path causes problems

### PR DESCRIPTION
**Pull request by andyherrick**
Fixing FileUtils.dirname() to skip over "/.".

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260335](https://bugs.openjdk.java.net/browse/JDK-8260335): [macos] Running app using relative path causes problems

### Reviewers
 * [Alexander Matveev](https://openjdk.java.net/census#almatvee) (@sashamatveev - Committer) ⚠️ Review applies to 6b5bfb28ff6a5706c4ed75f14f292f6bd2d33213
 * [Alexander Zuev](https://openjdk.java.net/census#kizune) (@azuev-java - **Reviewer**) ⚠️ Review applies to 6b5bfb28ff6a5706c4ed75f14f292f6bd2d33213

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2260/head:pull/2260`
`$ git checkout pull/2260`

### CODE REVIEW

**Review by sashamatveev — APPROVED** [review]

File: src/jdk.jpackage/share/native/common/FileUtils.cpp [hunk 1]
```diff
@@ -54,7 +53,15 @@ bool isDirSeparator(const tstring::value_type c) {
 
 
 tstring dirname(const tstring &path) {
-    tstring::size_type pos = path.find_last_of(_T("\\/"));
+    tstring::size_type pos;
+    if (tstrings::endsWith(path, _T("/."))) {
```
Comment (alexeysemenyukoracle) [inline src/jdk.jpackage/share/native/common/FileUtils.cpp:57]: I'd add `|| tstrings::endsWith(path, _T("\\."))` to cover similar case for Windows paths

**Review by alexeysemenyukoracle — CHANGES_REQUESTED** [review]

**Review by azuev-java — APPROVED** [review]
Aside of small change requested by @alexeysemenyukoracle looks fine.

**Pull request closed without merging**